packages/peregrine/src/hooks/useCarousel.js

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useState, useMemo } from 'react';
+import { useSortedImages } from './useSortedImages';
+
+/**
+ * A hook for interacting with the state of a carousel of images.
+ *
+ * @param {Array} images an array of image objects
+ * @param {number} startIndex the index at which to start the carousel
+ */
+export const useCarousel = (images, startIndex = 0) => {
+    const sortedImages = useSortedImages(images);
```

Comment:
I want to talk about making this a bit more generic. For starters, `useSortedImages` can just be a sort function passed to `useCarousel`.  `useCarousel` also doesn't have to be specific to images though the prop naming here suggests it is. It's just a circular linked list right now.